lib/gitlab_git/repository.rb

```diff
@@ -807,6 +807,32 @@ def autocrlf=(value)
         rugged.config['core.autocrlf'] = AUTOCRLF_VALUES.invert[value]
       end
 
+      # Return result like "git ls-files" , recursive and full file path only
+      #
+      # Ex.
+      #   repo.ls_files('master')
+      #
+      def ls_files(ref)
+        actual_ref = ref || root_ref
+
+        begin
+          sha = sha_from_ref(actual_ref)
+        rescue Rugged::OdbError, Rugged::InvalidError, Rugged::ReferenceError
+          # Return an empty array if the ref wasn't found
+          return []
+        end
+
+        cmd = %W(git --git-dir=#{path} ls-tree)
+        cmd += %W(-r)
+        cmd += %W(--full-tree)
+        cmd += %W(--full-name)
+        cmd += %W(--name-only)
+        cmd += %W(-- #{actual_ref})
+        raw_output = IO.popen(cmd) {|io| io.read }
```

Comment:
Space between { and | missing.<br>Pass `&:read` as an argument to `popen` instead of a block.